### DISCUSSION TITLE
How to set multiple native filters with the Javascript OnClick HREF?

### DISCUSSION THREAD

**Discussion by ArmadilloDisco** [apache/superset#24116] [Q&A / Help] [answered]
I am attempting to figure out how to apply multiple filters to a dashboard that is linked to by a map chart (made from the sf_population_polygon sample data).

I have successfully gotten a single filter to work using the the zipcode column:
`d => "/superset/dashboard/Map_Link_Filtering_2/?" + "native_filters=(NATIVE_FILTER-ZLzfix0V5:(__cache:(label:'" + d.object.extraProps.zipcode + "',validateStatus:!f,value:!('" + d.object.extraProps.zipcode + "')),extraFormData:(filters:!((col:'zipcode',op:IN,val:!('" + d.object.extraProps.zipcode + "')))),filterState:(label:'" + d.object.extraProps.zipcode + "',validateStatus:!f,value:!('" + d.object.extraProps.zipcode + "')),id:NATIVE_FILTER-ZLzfix0V5,ownState:()))" `

This works fine, but for testing I would like to also use the population of the clicked zipcode to filter the next dashboard as well (yes I know with that data it doesn't make sense to do that but this is more about figuring out how to use the feature).

I have attempted a few ways of joining the two:
`d => "/superset/dashboard/Map_Link_Filtering_2/?" + "native_filters=(NATIVE_FILTER-ZLzfix0V5:(__cache:(label:'" + d.object.extraProps.zipcode + "',validateStatus:!f,value:!('" + d.object.extraProps.zipcode + "')),extraFormData:(filters:!((col:'zipcode',op:IN,val:!('" + d.object.extraProps.zipcode + "')))),filterState:(label:'" + d.object.extraProps.zipcode + "',validateStatus:!f,value:!('" + d.object.extraProps.zipcode + "')),id:NATIVE_FILTER-ZLzfix0V5,ownState:()))" + "%2C(NATIVE_FILTER-Lt_WHq3Bi:(__cache:(label:'" + d.object.extraProps.population + "',validateStatus:!f,value:!('" + d.object.extraProps.population + "')),extraFormData:(filters:!((col:'population ',op:IN,val:!('" + d.object.extraProps.population + "')))),filterState:(label:'" + d.object.extraProps.population + "',validateStatus:!f,value:!('" + d.object.extraProps.population + "')),id:NATIVE_FILTER-Lt_WHq3Bi,ownState:()))"`

With comma, ampersand, plus sign, and their url encoded equivalents. all just break the filtering and link to the dashboard with no filters applied. 

How do I join the two native filter sets?

**Comment by ArmadilloDisco** [comment, marked as answer]
Through more trial and error, it turns out that you need to include them in both in same primary parenthesis, separated by a comma 

`d => "/superset/dashboard/Map_Link_Filtering_2/?" + "native_filters=(NATIVE_FILTER-ZLzfix0V5:(__cache:(label:'" + d.object.extraProps.zipcode + "',validateStatus:!f,value:!('" + d.object.extraProps.zipcode + "')),extraFormData:(filters:!((col:'zipcode',op:IN,val:!('" + d.object.extraProps.zipcode + "')))),filterState:(label:'" + d.object.extraProps.zipcode + "',validateStatus:!f,value:!('" + d.object.extraProps.zipcode + "')),id:NATIVE_FILTER-ZLzfix0V5,ownState:())" + "," + "NATIVE_FILTER-Lt_WHq3Bi:(__cache:(label:'" + d.object.extraProps.population + "',validateStatus:!f,value:!('" + d.object.extraProps.population + "')),extraFormData:(filters:!((col:'population ',op:IN,val:!('" + d.object.extraProps.population + "')))),filterState:(label:'" + d.object.extraProps.population + "',validateStatus:!f,value:!('" + d.object.extraProps.population + "')),id:NATIVE_FILTER-Lt_WHq3Bi,ownState:()))"`

Also here is a python code to generate this for the href, and a function to build out the CONCAT statement in a custom column
[SuperSet_Link_Builder.txt](https://github.com/apache/superset/files/11554361/SuperSet_Link_Builder.txt)

**Comment by jim1982ha** [comment]
this script works great!
if that's not too much to ask, can you help completing it to account the following cases:

1/ allow the generated URL to also pass the time filters to the target dashboard, if possible ? (say the common variable in source/target dashboard is called 'local_time')
2/ consider the case where there are multiples values selected for one of the configured dimensions in the script (say there are 3 values of dimension 'population' pre-filtered in the source dashboard => so far, it seems like only one (the 1st?) value is considered in the URL).

Thanks!
